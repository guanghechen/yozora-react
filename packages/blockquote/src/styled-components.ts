import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
