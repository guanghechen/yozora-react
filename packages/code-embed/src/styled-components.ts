import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
