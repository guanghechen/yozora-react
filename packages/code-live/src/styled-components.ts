import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
