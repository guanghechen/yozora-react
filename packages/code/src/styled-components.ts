import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
