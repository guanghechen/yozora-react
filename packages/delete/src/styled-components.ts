import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
