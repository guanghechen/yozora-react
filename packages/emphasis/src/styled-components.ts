import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
