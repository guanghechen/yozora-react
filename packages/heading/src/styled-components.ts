import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
