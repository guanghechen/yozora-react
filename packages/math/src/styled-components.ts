import type { YozoraMathTheme } from './theme'
import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    math?: YozoraMathTheme
  }
}
