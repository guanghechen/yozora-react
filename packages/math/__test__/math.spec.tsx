import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import Math from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render with custom className', () => {
    const code = 'x^2 + y^2 = z^2'
    const className = 'custom-math'
    const wrapper = render(
      <Math className={ className } value={ code } />
    )
    expect(wrapper.hasClass(className)).toEqual(true)
  })

  it('value is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<Math value={ value } />)
      }).toThrow(/The prop `value` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLDivElement>()
    const wrapper = mount(
      <Math ref={ ref } data-value="waw" value="" />
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const code = `
      \\begin{align}
        f(x) = \\left\\lbrace
          \\begin{aligned}
            &x^2, &x < 0 \\\\
            &\\frac{1}{x^3}, &x > 0
          \\end{aligned}
        \\right.
      \\end{align}
    `

    const wrapper = mount(
      <Math
        value={ code }
        style={{ color: 'orange', fontSize: '16px' }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        math: {
          padding: '2px',
          border: '1px solid blue',
          // margin: '0 2px',
          background: 'hsla(210deg, 13%, 12%, 0.05)',
          color: '#d81848',
        }
      }
    }

    const code = `
      \\begin{align}
        f(x) = \\left\\lbrace
          \\begin{aligned}
            &x^2, &x < 0 \\\\
            &\\frac{1}{x^3}, &x > 0
          \\end{aligned}
        \\right.
      \\end{align}
    `

    const wrapper = mount(
      <ThemeProvider theme={ theme }>
        <Math
          value={ code }
          style={{ color: 'orange', fontSize: '16px' }}
        />
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
