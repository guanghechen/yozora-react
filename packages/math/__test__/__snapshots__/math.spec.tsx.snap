// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot custom 1`] = `
<YozoraMath
  className="custom-class"
  style={
    Object {
      "background": "hsla(210deg, 13%, 12%, 0.05)",
      "border": "1px solid blue",
      "color": "#d81848",
      "margin": "0 2px",
      "padding": "2px",
    }
  }
  value="
\\\\begin{align}
  f(x) = \\\\left\\\\lbrace
    \\\\begin{aligned}
      &x^2, &x < 0 \\\\\\\\
      &\\\\frac{1}{x^3}, &x > 0
    \\\\end{aligned}
  \\\\right.
\\\\end{align}
"
>
  <div
    className="yozora-math custom-class"
    style={
      Object {
        "background": "hsla(210deg, 13%, 12%, 0.05)",
        "border": "1px solid blue",
        "color": "#d81848",
        "margin": "0 2px",
        "padding": "2px",
      }
    }
  >
    <MathJaxNode
      formula="
\\\\begin{align}
  f(x) = \\\\left\\\\lbrace
    \\\\begin{aligned}
      &x^2, &x < 0 \\\\\\\\
      &\\\\frac{1}{x^3}, &x > 0
    \\\\end{aligned}
  \\\\right.
\\\\end{align}
"
      inline={false}
    />
  </div>
</YozoraMath>
`;

exports[`snapshot default 1`] = `
<YozoraMath
  style={
    Object {
      "color": "orange",
      "fontSize": "16px",
    }
  }
  value="
\\\\begin{align}
  f(x) = \\\\left\\\\lbrace
    \\\\begin{aligned}
      &x^2, &x < 0 \\\\\\\\
      &\\\\frac{1}{x^3}, &x > 0
    \\\\end{aligned}
  \\\\right.
\\\\end{align}
"
>
  <div
    className="yozora-math"
    style={
      Object {
        "color": "orange",
        "fontSize": "16px",
      }
    }
  >
    <MathJaxNode
      formula="
\\\\begin{align}
  f(x) = \\\\left\\\\lbrace
    \\\\begin{aligned}
      &x^2, &x < 0 \\\\\\\\
      &\\\\frac{1}{x^3}, &x > 0
    \\\\end{aligned}
  \\\\right.
\\\\end{align}
"
      inline={false}
    />
  </div>
</YozoraMath>
`;
